pkg/Library/CoreXenon/Framework/Binder.js

```diff
@@ -3,158 +3,105 @@
-use `inputBindings` to find the Atom properties to update.
-*/
-const addInputBinding = (b$, key, id, prop) => (b$.inputBindings[key] ??= []).push({id, prop});
-
-/*
-outputBindings :: <layerScopeAtomKey>: {<prop>: <graphScopeKey>}
-
-If Atom `<layerScopeAtomKey>` produces output, the output fields 
-are matched against `prop`s in `outputBindings` to find the
-`<graphScopeKey>`s to update.
-*/
-const addOutputBinding = (b$, key, prop, binding) => (b$.outputBindings[key] ??= create(null))[prop] = binding;
-
 export const constructBindings = system => {
   const bindings = {
-    // mapping data keys from App-scope to Atom-scope
-    inputBindings: create(null),
-    // mapping data keys from Atom-scope to App-scope
-    outputBindings: create(null)
+    // maps state to atom inputs
+    input: create(null),
+    // maps atom outputs to state
+    output: create(null)
   };
   // system is a map of AtomSpecs, usually prepared from one or more Nodes, that
   // have been disambiguated to fit in one scope
   map(system, (atomId, atomSpec) => addAtomSpecBindings(bindings, atomId, atomSpec));
   // log
-  log.groupCollapsed('bindings:');
+  // log.groupCollapsed('bindings:');
   log(bindings);
-  log.groupEnd();
+  // log.groupEnd();
   // result
   return bindings;
 };
 
 const addAtomSpecBindings = (bindings, atomId, atomSpec) => {
-  // unpack the objectId
-  const objectId = Id.sliceId(atomId, 0, -1);
-  // add bindings from the atom spec
-  map(atomSpec.bindings, (prop, bound) => addBoundProperty(bindings, objectId, atomId, prop, bound));  
-  // inputs
-  atomSpec.inputs?.forEach(prop => addInputBinding(bindings, quid(atomId, prop), atomId, prop));
-  // outputs
-  atomSpec.outputs?.forEach(prop => addOutputBinding(bindings, atomId, prop, quid(atomId, prop)));
-};
-
-const addBoundProperty = (bindings, objectId, atomId, inProp, bound) => {
-  // Array-ify simple-string 'bound' values
-  bound = typeof bound === 'string' ? [bound] : bound;
-  // for each value bound on this atom spec
-  bound?.forEach(binding => {
-    // create mapping from qualified state prop to input prop
-    addInputBinding(bindings, quid(objectId, binding), atomId, inProp);
-    // create mapping from output prop to qualified state prop
-    const [outId, outProp] = Id.splitId(binding);
-    addOutputBinding(bindings, quid(objectId, outId), outProp, quid(objectId, binding));
+  // each atom input receives from a state key
+  atomSpec.inputs?.forEach(prop => {
+    const key = Id.qualifyId(atomId, prop);
+    addBinding(bindings.input, key, key);
   });
-};
-
-export const addConnections = (layerId, connections, inputBindings) => {
-  // update input/outputBindings with cross-node connections
-  entries(connections).forEach(([propId, bound]) => {
-    const {id, prop} = Id.parsePropId(propId);
-    if (typeof bound === 'string') {
-      bound = [bound];
+  // unpack the node id
+  const nodeId = Id.sliceId(atomId, 0, -1);
+  // add input bindings from the atom spec
```

Comment:
this bit has to deal with _implicit properties_

Reply:
👍